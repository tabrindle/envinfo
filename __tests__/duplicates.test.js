const envinfo = require('../src/envinfo');
const path = require('path');

describe('Running the programmatic interface', () => {
  test('return expected duplicates in json', async () => {
    const cwd = process.cwd();
    process.chdir(path.join(__dirname, 'packages/test-duplicates'));
    try {
      const data = await envinfo.run({ npmPackages: ['b'] }, { duplicates: true, json: true });
      expect(JSON.parse(data)).toMatchSnapshot();
    } finally {
      process.chdir(cwd);
    }
  });

  test('return expected duplicates in yaml', async () => {
    const cwd = process.cwd();
    process.chdir(path.join(__dirname, 'packages/test-duplicates'));
    try {
      const data = await envinfo.run({ npmPackages: ['b'] }, { duplicates: true });
      expect(data).toMatchSnapshot();
    } finally {
      process.chdir(cwd);
    }
  });
});
