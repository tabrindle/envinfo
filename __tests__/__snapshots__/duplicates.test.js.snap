// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Running the programmatic interface return expected duplicates in json 1`] = `
Object {
  "npmPackages": Object {
    "b": Object {
      "duplicates": Array [
        "1.0.2",
      ],
      "installed": "1.0.0",
      "wanted": "1.0.0",
    },
  },
}
`;

exports[`Running the programmatic interface return expected duplicates in yaml 1`] = `
"
  npmPackages:
    b: 1.0.0 => 1.0.0 (1.0.2)
"
`;
